arrows/vxl/average_frames.cxx

```diff
@@ -0,0 +1,770 @@
+}
+
+// ----------------------------------------------------------------------------
+average_frames
+::~average_frames()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+average_frames
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "type", averager_converter().to_string( d->type ),
+                     "Operating mode of this filter, possible values: " +
+                     averager_converter().element_name_string() );
+  config->set_value( "window_size", d->window_size,
+                     "The window size if computing a windowed moving average." );
+  config->set_value( "exp_weight", d->exp_weight,
+                     "Exponential averaging coefficient if computing an exp average." );
+  config->set_value( "round", d->round,
+                     "Should we spend a little extra time rounding when possible?" );
+  config->set_value( "output_variance", d->output_variance,
+                     "If set, will compute an estimated variance for each pixel which "
+                     "will be outputted as either a double-precision or byte image." );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+average_frames
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Starting with our generated vital::config_block to ensure that assumed
+  // values are present. An alternative is to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for averaging
+  d->type = config->get_enum_value< averager_converter >( "type" );
+  d->window_size = config->get_value< unsigned >( "window_size" );
+  d->exp_weight = config->get_value< double >( "exp_weight" );
+  d->round = config->get_value< bool >( "round" );
+  d->output_variance = config->get_value< bool >( "output_variance" );
+}
+
+// ----------------------------------------------------------------------------
+bool
+average_frames
+::check_configuration( vital::config_block_sptr config ) const
+{
+  auto const& type = config->get_enum_value< averager_converter >( "type" );
+  if( !( type == AVERAGER_cumulative ||
+         type == AVERAGER_window ||
+         type == AVERAGER_exponential ) )
+  {
+    return false;
+  }
+  else if( type == AVERAGER_exponential )
+  {
+    double exp_weight = config->get_value< double >( "exp_weight" );
+    if( exp_weight <= 0 || exp_weight > 1 )
+    {
+      return false;
+    }
+  }
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+kwiver::vital::image_container_sptr
+average_frames
+::filter( kwiver::vital::image_container_sptr image_data )
+{
+  // Perform Basic Validation
+  if( !image_data )
+  {
+    return image_data;
```

Comment:
Should there be logging of some kind here for this case?

Reply:
Updated, thanks.